include/pipettes/core/eeprom.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <array>
+#include <concepts>
+#include <cstdint>
+
+#include "common/core/bit_utils.hpp"
+
+/*
+ * A template and helper functions describing
+ */
+
+namespace i2c {
+template <class I2C>
+concept EEPromPolicy =
+    requires(I2C i2c_comms, const uint8_t transmit,
+             std::array<uint8_t, I2C::BUFFER_SIZE>& receive) {
+    {i2c_comms.BUFFER_SIZE};
+    {i2c_comms.transmit(transmit)};
```

Comment:
On lines 19-20, I am specifying that my concept requires a transmit/receive function. This allows me do `writer.transmit` and `writer.receive` where writer is my EEPromPolicy concept. @sotaoverride